### DISCUSSION TITLE
Are the contents of reaction file and reactionabcd file equaivalent?

### DISCUSSION THREAD

**Discussion by luzujiall** [deepmodeling/reacnetgenerator#2284] [Q&A]
This question was not explicitly stated in the manual, and I initially speculated that they were equivalent. However, after testing, I became skeptical about that。Specifically, I found that the count of the reaction "[H] [H] [N]=[N] ->[H] [H]" in the reaction file is not equal to the count of "[H] [H] [N]=[N] ->[H]+[N] # [N]" plus "[H] [H] [N]=[N] ->[H]+[N]=[N]" in the abcd file. What is the relationship between the two reaction files? Complementary?

**Comment by njzjz** [comment]
Assume there is a reaction `[H1] [H2] [N1]=[N2] + [H3] [H4] -> [H1] [H3] [N1]=[N2] + [H2] [H4]` (here 1, 2, 3, and 4 represent the index of atoms) where `[H2]` and `[H3]` exchange, `[H] [H] [N]=[N] ->[H] [H]` will be counted in the reaction file (on the atomic level), but no reactions will be shown in the abcd file as it cannot be defined as a chemical reaction.

**Reply by njzjz** [reply]
I assume that you have turned off the HMM...

**Reply by luzujiall** [reply]
> Assume there is a reaction `[H1] [H2] [N1]=[N2] + [H3] [H4] -> [H1] [H3] [N1]=[N2] + [H2] [H4]` (here 1, 2, 3, and 4 represent the index of atoms) where `[H2]` and `[H3]` exchange, `[H] [H] [N]=[N] ->[H] [H]` will be counted in the reaction file (on the atomic level), but no reactions will be shown in the abcd file as it cannot be defined as a chemical reaction.

Thank you very much for your reply. So the reaction file records all atomic exchange behaviors, while the abcd file records all species change behaviors, is that correct？
